Sources/PostmarkClient/MailClient.swift

```diff
@@ -0,0 +1,74 @@
+import HTTP
+import Mail
+import Vapor
+
+
+public final class MailClient {
+    
+    let serverToken: String
+    let client: ClientProtocol
+    public var result : Response?
+    
+    public init(clientProtocol: ClientProtocol.Type, serverToken: String) throws {
+        self.serverToken = serverToken
+        client = try clientProtocol.make(scheme: "https", host: "api.postmarkapp.com")
+    }
+}
+
+extension EmailAddress {
+    var smtpLongFormatted: String {
+        var formatted = ""
+        
+        if let name = self.name {
+            formatted += name
+            formatted += " "
+        }
+        formatted += "<\(address)>"
+        return formatted
+    }
+}
+
+extension Sequence where Iterator.Element == EmailAddress {
+    var smtpLongFormatted: String {
+        return self.map { $0.smtpLongFormatted } .joined(separator: ", ")
+    }
+}
+
+
+extension MailClient: MailClientProtocol {
+    
+    public convenience init() throws {
+        guard let clientProtocol = Provider.clientProtocol else {
+            throw Provider.Error.noClient
+        }
+        guard let serverToken = Provider.serverToken else {
+            throw Provider.Error.missingConfig("serverToken")
+        }
+        try self.init(clientProtocol: clientProtocol, serverToken: serverToken)
+    }
+    
+    public func send(_ emails: [Mail.Email]) throws {
+        
+        try emails.forEach { email in
+
+            let jsonEmail = JSON([
+                "From": email.from.emailAddress.smtpLongFormatted.makeNode(),
+                "To": email.to.smtpLongFormatted.makeNode(),
+                "Subject": email.subject.makeNode(),
+                "HtmlBody": email.body.content.makeNode(),
+            ])
+            
+            let headers : [HeaderKey: String] = [
+                "Content-Type": "application/json",
+                "Accept": "application/json",
+                "X-Postmark-Server-Token": serverToken
+            ]
+            
+            result = try client.post(
```

Comment:
Postmark has a [list of HTTP response codes](http://developer.postmarkapp.com/developer-api-overview.html#response-codes).

I would `throw` on any unexpected (non-`200`) response status code. There are four expected error codes: `401`, `422`, `500`, `503`. I would create a `PostmarkError` enum with cases to cover each of these results, plus one for any unexpected result code from Postmark.

The enum member for `422` should have an associated value in which the [API error code](http://developer.postmarkapp.com/developer-api-overview.html#error-codes) can be stored and logged.

An alternative would be to group related API error codes together, and add an enum case for each such group. For example, have one case for all syntax errors, one group for all configuration errors, one group for billing errors.